web/packages/teleport/src/JoinTokens/CreateJoinTokenDialog.tsx

```diff
@@ -0,0 +1,568 @@
+  updateTokenList: (token: JoinToken) => void;
+  editToken?: JoinToken;
+  editTokenWithYAML: (tokenId: string) => void;
+}) => {
+  const ctx = useTeleport();
+  const [newTokenState, setNewTokenState] = useState<NewJoinTokenState>(
+    editToken ? makeDefaultEditState(editToken) : defaultNewTokenState
+  );
+
+  const [createTokenAttempt, runCreateTokenAttempt] = useAsync(
+    async (req: CreateJoinTokenRequest) => {
+      const token = await ctx.joinTokenService.createJoinToken(req);
+      updateTokenList(token);
+      onClose();
+    }
+  );
+
+  function reset(validator) {
+    validator.reset();
+    setNewTokenState(defaultNewTokenState);
+  }
+
+  async function save(validator) {
+    if (!validator.validate()) {
+      return;
+    }
+
+    const request: CreateJoinTokenRequest = {
+      name: newTokenState.name,
+      roles: newTokenState.roles.map(r => r.value),
+      join_method: newTokenState.method.value,
+    };
+
+    if (newTokenState.method.value === 'iam') {
+      request.allow = newTokenState.iam;
+    }
+
+    if (request.roles.includes('Bot')) {
+      request.bot_name = newTokenState.bot_name;
+    }
+
+    if (newTokenState.method.value === 'gcp') {
+      const gcp = {
+        allow: newTokenState.gcp.map(rule => ({
+          project_ids: rule.project_ids?.map(id => id.value),
+          locations: rule.locations?.map(loc => loc.value),
+          service_accounts: rule.service_accounts?.map(
+            account => account.value
+          ),
+        })),
+      };
+      request.gcp = gcp;
+    }
+
+    runCreateTokenAttempt(request);
+  }
+
+  function setTokenRoles(roles: Option<JoinRole, JoinRole>[]) {
+    setNewTokenState(prevState => ({
+      ...prevState,
+      roles,
+    }));
+  }
+
+  function setTokenMethod(method: Option<JoinMethod, JoinMethod>) {
+    // set the method and reset the token rules per type for a fresh form
+    setNewTokenState(prevState => ({
+      ...prevState,
+      method,
+      iam: [{ aws_account: '', aws_arn: '' }], // default
+    }));
+  }
+
+  function setTokenField(fieldName: string, value: string) {
+    setNewTokenState(prevState => ({
+      ...prevState,
+      [fieldName]: value,
+    }));
+  }
+
+  return (
+    <Flex width="500px">
+      <Box
+        pr={4}
+        pl={4}
+        css={`
+          overflow: auto;
+          width: 100%;
+          min-height: 50vh;
+          padding-bottom: 0;
+        `}
+      >
+        <Flex
+          alignItems="center"
+          mb={3}
+          justifyContent="space-between"
+          maxWidth={maxWidth}
+        >
+          <Flex alignItems="center" mr={3}>
+            <HoverTooltip tipContent="Back to Join Tokens">
+              <ButtonIcon onClick={onClose} mr={2} ml={'-8px'}>
+                <Cross size="medium" />
+              </ButtonIcon>
+            </HoverTooltip>
+            <Text typography="h3" fontWeight={400}>
+              {editToken ? `Edit Token` : 'Create a New Join Token'}
```

Comment:
Here the button would be "Create a New Join Token" and elsewhere there's a button that's "Create new token" - we should be consistent on capitalisation